### PR TITLE
Feature/search page

### LINKED ISSUE #43: Landing search and results page

### PR DESCRIPTION
**Pull request by vivekimsit**
I wanted to re-use house components made by @nikrb but I felt they are not generic enough.
Lets refactor them later.

closes #43

### CODE REVIEW

File: frontend/src/components/Login/Login.js [hunk 1]
```diff
@@ -7,84 +7,67 @@ import axios from 'axios';
...
-		this.setState({email: "", password: ""});
-	}
-
-
-
-	render() {
-
-		return (
-
+  login= () => {
```
Comment (Mortuie) [inline frontend/src/components/Login/Login.js]: space missing.

File: frontend/src/containers/Search/SearchPage.js [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+    height: '80vh',
+    width: '100%',
+    display: 'flex',
+    justifyContent: 'center',
+    alignItems: 'center',
+    backgroundImage: "url('http://lorempixel.com/1000/600/abstract')",
+    backgroundSize: 'cover',
+    backgroundPosition: 'center',
+    overflow: 'scroll',
+    marginTop: '0'
```
Comment (Mortuie) [inline frontend/src/containers/Search/SearchPage.js]: `marginTop: 0`

File: frontend/src/routes/Routes.js [hunk 3]
```diff
@@ -24,8 +25,7 @@ class routes extends Component {
     const {user} = this.props;
     return (
       <Switch>
-
-        <Route exact path="/" component={Dummy} />
+        <Route path="/" component={SearchLayout} />
```
Comment (Mortuie) [inline frontend/src/routes/Routes.js]: Why did you remove the exact.... ?

**Review by Mortuie — DISMISSED** [review]

File: frontend/src/components/House/House.js [hunk 4]
```diff
@@ -4,18 +4,25 @@ import styles from './styles';
 
 export default class House extends Component {
   onNewIssue = () => {
-    this.props.onNewIssue( this.props.data);
```
Comment (nikrb) [inline frontend/src/components/House/House.js]: I have fixed some stuff like this in my review changes, also moved inline styles to aphrodite. I have also refactored this component to use a Card using your css ... merge will be fun :)

File: frontend/src/containers/House/House.js [hunk 5]
```diff
@@ -9,13 +9,16 @@ class House extends Component {
   componentWillMount = () => {
     this.setState( {new_house: this.props.house});
   };
+
   componentWillReceiveProps = (nextProps) => {
     this.setState( { new_house: nextProps.house});
   };
-  houseFormSubmit = e => {
+
+  houseFormSubmit = e => { // eslint-disable-line no-unused-vars
```
Comment (nikrb) [inline frontend/src/containers/House/House.js]: good catch. I would prefer to just leave the "e" out and have `= () => {` rather than disable rules? Any reason for keeping an unused param @vivekimsit ?

File: frontend/src/containers/Search/Background.js [hunk 6]
```diff
@@ -0,0 +1,15 @@
+import React from 'react';
+import { css, StyleSheet } from 'aphrodite';
+
+export default () => (
+  <img src="http://lorempixel.com/1000/600/abstract" className={css(styles.img)} alt="background" />
```
Comment (nikrb) [inline frontend/src/containers/Search/Background.js]: I find this very busy, maybe I'm just old ^^   It took a few seconds to load and I think we have a more minimalist style currently? just my two pennies.

File: frontend/src/containers/Search/SearchPage.js [hunk 7]
```diff
@@ -0,0 +1,62 @@
...
+    const postCode = this.getInputValue();
+    this.props.history.push(`search?postCode=${postCode}`);
+  }
+
+  render () {
+    return (
+      <div className={css(styles.container)}>
+        <Form onSubmit={this.handleSubmit}>
+          <input type="search"
+                 className={css(styles.input)}
```
Comment (nikrb) [inline frontend/src/containers/Search/SearchPage.js]: indent two spaces? Oh I see you've lined up the attribs, is that normal? Looks good, I can do that :)

File: frontend/src/containers/Search/actions.js [hunk 8]
```diff
@@ -0,0 +1,37 @@
...
+    postCode
+  };
+}
+
+export const SEARCH_HOUSES_SUCCESS = 'SEARCH_HOUSES_SUCCESS';
+export function searchHousesSuccess(postCode, json) {
+  return {
+    type: SEARCH_HOUSES_SUCCESS,
+    postCode,
+    houses: (json.houses || [])
```
Comment (nikrb) [inline frontend/src/containers/Search/actions.js]: I don't think actions should manipulate data? Is this because you're just doing the filtering locally for now? shouldn't it be in the reducer anyway? These are more questions than opinion, really not confident with redux myself yet.

**Review by nikrb — APPROVED** [review]

File: frontend/src/containers/House/House.js (hunk 5 above)
Comment (vivekimsit) [reply frontend/src/containers/House/House.js]: I have no idea but you maybe interested in using the event object later? I thought not to touch your code.

File: frontend/src/containers/Search/Background.js (hunk 6 above)
Comment (vivekimsit) [reply frontend/src/containers/Search/Background.js]: Yes, it was just for fun. I will see if there is some good image hosting CDN or we may save the background image in our server.

File: frontend/src/containers/Search/SearchPage.js (hunk 7 above)
Comment (vivekimsit) [reply frontend/src/containers/Search/SearchPage.js]: I looked good to my eyes ;)

File: frontend/src/containers/Search/actions.js (hunk 8 above)
Comment (vivekimsit) [reply frontend/src/containers/Search/actions.js]: I read couple of articles and it made some sense to me i.e. "Action can be fat but reducers should be lean", the rationale was reducers are just pure functions and it should not be responsible for business logic.

File: frontend/src/routes/Routes.js (hunk 3 above)
Comment (vivekimsit) [reply frontend/src/routes/Routes.js]: I agree `exact` should be there.

File: frontend/src/components/House/House.js (hunk 4 above)
Comment (vivekimsit) [reply frontend/src/components/House/House.js]: I will wait for your PR to be merged first then.

File: frontend/src/containers/Search/actions.js (hunk 8 above)
Comment (nikrb) [reply frontend/src/containers/Search/actions.js]: I'm only learning redux, but I thought that was exactly the opposite way round? From the redux dox:
```
Reducers specify how the application's state changes in response to actions sent to the store. Remember that actions only describe the fact that something happened, but don't describe how the application's state changes.
```

So I'm completely confused now ... can you post a link to the articles you have read?

File: frontend/src/components/House/House.js (hunk 4 above)
Comment (nikrb) [reply frontend/src/components/House/House.js]: unfortunately my refactor is still two pr's away. Don't worry I'll sort it ... I do have more time than you after all ;)

File: frontend/src/containers/Search/actions.js (hunk 8 above)
Comment (vivekimsit) [reply frontend/src/containers/Search/actions.js]: Hmm, makes sense. I must have read it somewhere, too much to digest I will research it more.
